### DISCUSSION TITLE
There is an option to dump to file current variables content?

### DISCUSSION THREAD

**Discussion by CA10C3SAR** [zero-plusplus/vscode-autohotkey-debug#121] [Q&A]
First, thank you for the new option: `useAutoJumpToError: true`
I did some tests and looks like it's working nicely.

Topic:
There is an option to dump to file or maybe even store it to a variable, the content of the current variables?
Something similar to this option:
![image](https://user-images.githubusercontent.com/6724006/130293200-60409bd3-65a1-4a61-84a1-e7b918ca2978.png)

Maybe:
; @Debug-Breakpoint(x == 100) Debug-VariablesContent txt.txt

**Comment by zero-plusplus** [comment]
**I use a DeepL translator.**

Thanks for trying out the new features. I'm glad you find it useful.

---
> There is an option to dump to file

I have opened a draft of a new feature to support this. You can join the discussion if you like.
* #122

---
> maybe even store it to a variable, the content of the current variables?

If I implement this feature, how will you use it?

**Reply by CA10C3SAR** [reply]
My English is also not so good, I'm reading the draft, I did not understand if you can save to file just one specific variable or all variables in the currently running script till the debug breakpoint.

> If I implement this feature, how will you use it?

I would use it together ListLines, I did a function that continuos dumps the last **X** executed lines and saves them to a file in that current time.

Then when I'm debugging I usually leave my screen being recorded, and when i need to check something i open the file and check the lines executed at that determined time. 
But i don't have yet a way to verify the value of all variables in that current specific time. I would use this option to it.

**Reply by zero-plusplus** [reply]
Implement `@Debug-Log` so that it can output variables in the same way as `@Debug-Output`.
For example, you can write the contents of a variable to a file with `@Debug-Log =>{variable}`.
However, the value to be written must be a primitive value such as a string or number. This is because dumping an object is very costly.

---

I'm starting to understand what you want to do.

AutoHotkey can have environment-dependent bugs that are hard to reproduce, and I can very much understand your desire to check the values of variables, etc. at the time the problem occurred. However, as a practical matter, it is difficult to implement a process that writes all variables to the log on the debugger side, and even if it were possible, the process would be very slow.

A practical solution would be for the debugger to provide the current variable names, and the AutoHotkey side to dump the variables. This would be a fast process, and using the JSON library, objects could be recorded correctly.

**Reply by zero-plusplus** [reply]
First of all, I don't speak English either, so communication may be difficult, but I will take as much time as I need, so please feel free to comment.

I decided that it was not appropriate for the debugger adapter to handle the log file, so I stopped implementing #122.

However, I decided to add a new [meta variable](https://github.com/zero-plusplus/vscode-autohotkey-debug/issues/151) that will output all variables of local or global, etc. to the debug console.

I will also add the feature to label information such as the time of the output and whether it is a variable of which function, so I think I can meet your request except for writing to the log file.
